### PR TITLE
Fix pipelines transient errors

### PR DESCRIPTION
**Pull request by YunchuWang**
## Description

Fixes #

---

### PR information

- [ ] The title of the PR is clear and informative.
- [ ] There are a small number of commits, each of which has an informative message. This means that previously merged commits do not appear in the history of the PR. For information on cleaning up the commits in your pull request, [see this page](https://github.com/Azure/azure-powershell/blob/master/documentation/development-docs/cleaning-up-commits.md).
- [ ] If applicable, the PR references the bug/issue that it fixes in the description.
- [ ] New Unit tests were added for the changes made and CI is passing.

### Quality of Code and Contribution Guidelines
- [ ] I have read the [contribution guidelines](https://github.com/Azure/azure-functions-python-worker/blob/master/.github/CONTRIBUTING.md).

### CODE REVIEW

File: .github/workflows/codeql-analysis.yml [hunk 1]
```diff
@@ -32,40 +32,39 @@ jobs:
     strategy:
       fail-fast: false
       matrix:
-        language: [ 'python' ]
+        language: ["python"]
         # CodeQL supports [ 'cpp', 'csharp', 'go', 'java', 'javascript', 'python', 'ruby' ]
         # Learn more about CodeQL language support at https://git.io/codeql-language-support
-    permissions: read-all
```
Comment (gavin-aguiar) [inline .github/workflows/codeql-analysis.yml:38]: Is removing the permissions intentional?

File: .github/workflows/ci_e2e_workflow.yml [hunk 2]
```diff
@@ -38,6 +38,10 @@ jobs:
         uses: actions/setup-dotnet@v1
         with:
           dotnet-version: '6.x'
```
Comment (gavin-aguiar) [inline .github/workflows/ci_e2e_workflow.yml:40]: If this is not used then lets remove the old dotnet versions

File: .github/workflows/codeql-analysis.yml [hunk 3]
```diff
@@ -32,40 +32,39 @@ jobs:
...
-    
+
     steps:
-    - name: Checkout repository
```
Comment (vrdmr) [inline .github/workflows/codeql-analysis.yml:41]: Why these changes?

File: tests/utils/testutils.py [hunk 4]
```diff
@@ -257,6 +257,11 @@ def setUpClass(cls):
                 _setup_func_app(TESTS_ROOT / script_dir)
                 cls.webhost = start_webhost(script_dir=script_dir,
                                             stdout=cls.host_stdout)
+            if cls.webhost._proc.errors is not None or \
+               cls.webhost._proc.stderr is not None:
+                raise RuntimeError(
+                    'Failed to start WebHost:'
```
Comment (gavin-aguiar) [inline tests/utils/testutils.py:263]: If both are true then we should print both, errors and stderr.

File: tests/utils/testutils.py [hunk 5]
```diff
@@ -257,6 +257,11 @@ def setUpClass(cls):
...
+                    f'{cls.webhost._proc.errors or cls.webhost._proc.stderr}')
```
Comment (vrdmr) [inline tests/utils/testutils.py:264]: Wouldn't stderr always be populated? As there could be some processes which would use the STDERR stream?

File: .github/workflows/codeql-analysis.yml (hunk 1 above)
Comment (YunchuWang) [reply .github/workflows/codeql-analysis.yml:38]: yes there is already permission defined above

File: .github/workflows/codeql-analysis.yml (hunk 3 above)
Comment (YunchuWang) [reply .github/workflows/codeql-analysis.yml:41]: auto format issues, will update

File: tests/utils/testutils.py (hunk 4 above)
Comment (YunchuWang) [reply tests/utils/testutils.py:263]: discussed with Varad, this wont catch anything, removed

File: .github/workflows/ci_e2e_workflow.yml (hunk 2 above)
Comment (YunchuWang) [reply .github/workflows/ci_e2e_workflow.yml:40]: not too certain, will address in next pr

**Review by vrdmr — APPROVED** [review]

**Review by gavin-aguiar — APPROVED** [review]
